pkg/api/management/api/alert/notifier_action.go

```diff
@@ -0,0 +1,376 @@
+package alert
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/mail"
+	"net/smtp"
+	"net/textproto"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/rancher/norman/parse"
+	"github.com/rancher/norman/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NotifierCollectionFormatter(apiContext *types.APIContext, collection *types.GenericCollection) {
+	collection.AddAction(apiContext, "send")
+}
+
+func NotifierFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+	resource.AddAction(apiContext, "send")
+}
+
+func (h *Handler) NotifierActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	switch actionName {
+	case "send":
+
+		return h.testNotifier(actionName, action, apiContext)
+	}
+
+	return errors.Errorf("bad action %v", actionName)
```

Comment:
Any reason for not throwing `httperror.NewAPIError(httperror.InvalidAction`?

Reply:
I copy this from user_action.go.  I will change it to httperror.NewAPIError(httperror.InvalidAction). Thanks